phpBB/phpbb/search/fulltext_sphinx.php

```diff
@@ -430,22 +435,27 @@ protected function config_generate()
 	*/
 	public function split_keywords(&$keywords, $terms)
 	{
+		// Keep quotes and new lines
+		$keywords = str_replace(['&quot;', "\n"], ['"', ' '], trim($keywords));
+
 		if ($terms == 'all')
 		{
-			$match		= array('#\sand\s#i', '#\sor\s#i', '#\snot\s#i', '#\+#', '#-#', '#\|#', '#@#');
-			$replace	= array(' & ', ' | ', '  - ', ' +', ' -', ' |', '');
+			// Replaces verbal operators OR and NOT with special characters | and -, unless appearing within quotation marks
+			$match		= ['#\sor\s(?=([^"]*"[^"]*")*[^"]*$)#i', '#\snot\s(?=([^"]*"[^"]*")*[^"]*$)#i'];
+			$replace	= [' | ', ' -'];
 
 			$keywords = preg_replace($match, $replace, $keywords);
 			$this->sphinx->SetMatchMode(SPH_MATCH_EXTENDED);
 		}
 		else
 		{
+			$match = ['\\', '(',')','|','!','@','~', '/', '^', '$', '=','&amp;', '&lt;', '&gt;'];
+			$replace = [' ', ' ', ' ', ' ',' ',' ',' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '];
+
+			$keywords = str_replace($match, $replace, $keywords);
```

Comment:
As you're replacing all of thise with an empty string, I think you should rather use this:
```suggestion
			$keywords = str_replace($match, ' ', $keywords);
```
And get rid of the `$replace` array.